tests/ui/unneeded_wildcard_pattern.rs

```diff
@@ -0,0 +1,41 @@
+// run-rustfix
+#![feature(stmt_expr_attributes)]
+#![deny(clippy::unneeded_wildcard_pattern)]
+
+fn main() {
+    let t = (0, 1, 2, 3);
+
+    if let (0, .., _) = t {};
+    if let (0, _, ..) = t {};
+    if let (0, _, _, ..) = t {};
+    if let (0, .., _, _) = t {};
+    if let (_, 0, ..) = t {};
+    if let (.., 0, _) = t {};
```

Comment:
`if let (_, .., 2, 3) = t {}` doesn't get linted, does it? Can you add a test for it? This could actually lead to refactoring errors, because with the `_, ..` it means, there is at least one element before the `2`.

Reply:
It absolutely will. This is the purpose of this lint.
What errors could this lead to and how is this different from the case `(0, .., _, _)` →`(0, ..)`?

Reply:
I can't remember what I had in mind. And I also cannot come up with something. Can you add a test for it though?